agent/engine/task_manager_linux.go

```diff
@@ -33,12 +43,57 @@ func (mtask *managedTask) CleanupPlatformResources() error {
 
 // setupCgroup sets up the cgroup for each managed task
 func (mtask *managedTask) setupCgroup() error {
-	// Stub for now
+	cgroupRoot, err := mtask.Task.BuildCgroupRoot()
+	if err != nil {
+		return errors.Wrap(err, "mtask setup cgroup: unable to determine cgroup root")
+	}
+
+	seelog.Debugf("Setting up cgroup at: %s", cgroupRoot)
+
+	if control.Exists(cgroupRoot) {
```

Comment:
Under which scenario will this happen, what if two tasks accidentally have the same `cgroupRoot`? Should this error just be ignored?

Reply:
Two tasks on the same instance should never have the same taskARN. 
This is primarily to avoid task cgroup recreation